test/e2e/external_oidc_test.go

```diff
@@ -2,54 +2,244 @@ package e2e
+	kcClient, idpName, c := test.AddKeycloakIDP(t, tc.kubeConfig, true)
+	cleanups = append(cleanups, c...)
+	t.Logf("keycloak Admin URL: %s", kcClient.AdminURL())
+
+	// default-ingress-cert is copied to openshift-config and used as the CA for the IdP
+	// see test/library/idpdeployment.go:332
+	caBundleName := idpName + "-ca"
+	idpURL := kcClient.IssuerURL()
+
+	// run tests
+
+	testSpec := authSpecForOIDCProvider(idpName, idpURL, caBundleName, oidcGroupsClaim, oidcClientId)
+
+	t.Run("auth-config cm must not exist and gets deleted by the CAO if manually created when type not OIDC", func(t *testing.T) {
+		_, err := tc.kubeClient.CoreV1().ConfigMaps(managedNS).Get(testCtx, authCM, metav1.GetOptions{})
+		require.True(t, errors.IsNotFound(err), "openshift-config-managed/auth-config configmap must be missing")
+
+		// create cm
+		cm := v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      authCM,
+				Namespace: managedNS,
+			},
+			Data: map[string]string{
+				"test": "value",
+			},
+		}
+		newCM, err := tc.kubeClient.CoreV1().ConfigMaps(managedNS).Create(testCtx, &cm, metav1.CreateOptions{})
+		require.NoError(t, err)
+		require.Equal(t, cm.Data, newCM.Data)
+
+		// wait for CAO to delete it
+		wait.PollUntilContextTimeout(testCtx, 2*time.Second, 1*time.Minute, false, func(ctx context.Context) (bool, error) {
+			_, err := tc.kubeClient.CoreV1().ConfigMaps(managedNS).Get(testCtx, authCM, metav1.GetOptions{})
+			if errors.IsNotFound(err) {
+				return true, nil
+			}
+
+			return false, err
+		})
+	})
+
+	t.Run("invalid OIDC config degrades auth operator", func(t *testing.T) {
+		for _, tt := range []struct {
+			name       string
+			specUpdate func(*configv1.AuthenticationSpec)
+		}{
+			{"invalid issuer CA bundle", func(s *configv1.AuthenticationSpec) {
+				s.OIDCProviders[0].Issuer.CertificateAuthority.Name = "invalid-ca-bundle"
+			}},
+			{"invalid issuer URL", func(s *configv1.AuthenticationSpec) {
+				s.OIDCProviders[0].Issuer.URL = "https://invalid-idp.testing"
+			}},
+		} {
+			t.Run(tt.name, func(t *testing.T) {
+				err := tc.authResourceRollback(testCtx, origAuthSpec)
+				require.NoError(t, err, "failed to roll back auth resource")
+
+				tc.updateAuthResource(t, testCtx, testSpec, tt.specUpdate)
+
+				require.NoError(t, test.WaitForClusterOperatorDegraded(t, tc.configClient.ConfigV1(), "authentication"))
+			})
+		}
+	})
+
+	t.Run("invalid OIDC config crashes kube-apiserver pods during rollout", func(t *testing.T) {
+		tc.updateAuthResource(t, testCtx, testSpec, func(baseSpec *configv1.AuthenticationSpec) {
+			baseSpec.OIDCProviders[0].ClaimMappings.Username.Claim = ""
+		})
+
+		require.NoError(t, test.WaitForClusterOperatorStatusAlwaysAvailable(t, testCtx, tc.configClient.ConfigV1(), "authentication"))
+		require.NoError(t, test.WaitForClusterOperatorStatusAlwaysProgressing(t, testCtx, tc.configClient.ConfigV1(), "kube-apiserver"))
+
+		tc.requireKASRolloutCrash(t, testCtx, nil, 0)
+	})
```

Comment:
I think we want to make sure we _don't_ crash the kube-apiserver wherever possible. 

I know that this would cause the e2e test to permafail, but I'd rather we not have a false sense of "success" and instead fix this gap. If you don't think it makes sense to fix this as part of this PR, that's fine - but I _do_ think we should assert what we actually want to happen here (i.e we go degraded instead of applying the config) and deal with the permafail there.

This might mean that this PR doesn't merge until after a fix of some kind is put in place, but I think that price is worth paying for us to not forget about this gap and push it under the rug.

Reply:
My intention was to fix this separately to avoid crashing KAS pods at all, and still have the test validate the current behavior. As discussed offline, we'll skip this test for now, and re-enable once the fix is in place.